### PR TITLE
4 get/articles

### PR DESCRIPTION
**Pull request by pzadev**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -76,3 +76,38 @@ describe("GET /api/articles/:article_id", () => {
...
 });
+
+describe("GET /api/articles", () => {
+  test("should respond with an articles array of article objects", () => {
+    return request(app)
+      .get("/api/articles")
+      .expect(200)
+      .then(({ body }) => {
+        const { articles } = body;
+        articles.forEach((article) => {
```
Comment (nc-mezz-davies) [inline __tests__/app.test.js:87]: It is important to include a length check (you know there are articles in the test DB) as if the array is empty the forEach will not run (so none of the assertions will be reached) and the test will give a false positive

File: models.js [hunk 2]
```diff
@@ -5,12 +5,31 @@ exports.fetchTopics = () => {
...
+    COUNT(comments.comment_id) AS comment_count
+    FROM articles 
+    LEFT JOIN comments ON articles.article_id = comments.article_id
+    GROUP BY articles.article_id
+    ORDER BY articles.created_at DESC;
+  `;
+
+  return db.query(queryText).then(({ rows }) => {
+    rows.map((article) => {
+        article.comment_count = Number(article.comment_count)
```
Comment (nc-mezz-davies) [inline models.js:20]: This can be achieved with sql so that the comment count is a number when fetched - look into CAST

**Review by nc-mezz-davies — COMMENTED** [review]
You were correct that there are no client side errors to handle for this endpoint. Looks good 😊
